ledger/accountdb.go

```diff
@@ -2592,7 +2593,12 @@ func accountsInitDbQueries(q db.Queryable) (*accountsDbQueries, error) {
 		return nil, err
 	}
 
-	qs.lookupKeysByPrefixStmt, err = q.Prepare("SELECT acctrounds.rnd, kvstore.key FROM acctrounds LEFT JOIN kvstore ON SUBSTR (kvstore.key, 1, ?) = ? WHERE id='acctbase'")
+	qs.lookupKeysByPrefixStmt, err = q.Prepare("SELECT acctrounds.rnd, kvstore.key FROM acctrounds LEFT JOIN kvstore ON kvstore.key >= ? AND kvstore.key < ? WHERE id='acctbase'")
```

Comment:
I haven't checked the entire PR yet, but why do we need both of these variants?

Reply:
I can't tell if this is what you're asking from the highlighted lines, but if the question is why we need the two different SELECT queries, it's to handle the case of a query of the prefix "0xFFFFFF" (or any all 0xff) query.  There's no acceptable "high end" to compare to.

(Of course, so a query is impossible given box format, but I think it was nice to make sure it worked no matter what.)

Reply:
The previous query check requires `SUBSTR`, and it caused the linear scan.

The current clause seems a little weird, it requires `kvstore.key >= prefix`, and `kvstore.key < prefix + 1`. In this way, we can still achieve finding key by prefix, without paying a O(n) cost. Most of the thinking are reflecting in the previous discussion in the channel.

Reply:
The corner case cannot be covered very well, if the prefix is of form all `0xFF`, so a new query statement is now needed, maybe (hopefully) not?

Reply:
Maybe the intention is a bit clearer under the following renaming?

* `qs.lookupKeysForPrefixCaseTypicalStmt`
* `qs.lookupKeysForPrefixCaseEmptyOrAllFsStmt`

Reply:
changed accordingly.